doc/source/raysgd/v2/user_guide.rst

```diff
@@ -126,42 +135,121 @@ Porting code to RaySGD
 
         TODO. Write about how to convert code to use horovod.
 
+.. To make existing code from the previous SGD API, see :ref:`Backwards Compatibility <sgd-backwards-compatibility>`.
 
+Configurations
+--------------
 
-Training on a large dataset
----------------------------
+With RaySGD, you can execute a training function (``train_func``) in a
+distributed manner by calling ``trainer.run(train_func)``. To pass arguments
+into the training function, you can expose a single ``config`` parameter:
 
-SGD provides native support for :ref:`Ray Datasets <datasets>`. You can pass in a Dataset to RaySGD via ``Trainer.run``.
-Underneath the hood, RaySGD will automatically shard the given dataset.
+.. code-block:: diff
 
+    -def train_func():
+    +def train_func(config):
+
+Then, you can pass in the config dictionary as an argument to ``Trainer.run``:
+
+.. code-block:: diff
+
+    -trainer.run(train_func)
+    +config = {} # This should be populated.
+    +trainer.run(train_func, config=config)
+
+Putting this all together, you can run your training function with different
+configurations. As an example:
 
 .. code-block:: python
 
+    from ray.util.sgd.v2 import Trainer
+
     def train_func(config):
-        batch_size = config["worker_batch_size"]
-        data_shard = ray.sgd.get_data_shard()
-        dataloader = data_shard.to_torch(batch_size=batch_size)
+        results = []
+        for i in range(config["num_epochs"]):
+            results.append(i)
+        return results
 
-        for x, y in dataloader:
-            output = model(x)
-            ...
+    trainer = Trainer(backend="torch", num_workers=2)
+    trainer.start()
+    print(trainer.run(train_func, config={"num_epochs": 2}))
+    # [[0, 1], [0, 1]]
+    print(trainer.run(train_func, config={"num_epochs": 5}))
+    # [[0, 1, 2, 3, 4], [0, 1, 2, 3, 4]]
+    trainer.shutdown()
 
-        return model
+A primary use-case for ``config`` is to try different hyperparameters. To
+perform hyperparameter tuning with RaySGD, please refer to the
+:ref:`Ray Tune integration <tune-sgd>`.
 
-    trainer = Trainer(num_workers=8, backend="torch")
-    dataset = ray.data.read_csv("...").filter().pipeline(length=50)
+.. TODO add support for with_parameters
 
-    result = trainer.run(
-        train_func,
-        config={"worker_batch_size": 64},
-        dataset=dataset)
+.. _sgd-logging:
 
+Logging
```

Comment:
I added a line for custom logging callback within the `Logging Callbacks` section. Since this section already links to `Logging Callbacks`, I didn't add another section here.

For `run_iterator` I left a TODO since that section itself is still commented out in the docs.

Reply:
Can we also add a TODO for showing an example of a custom callback?

Reply:
What type of example did you have in mind, and where did you want it to be? I added some notes in `Logging Callbacks` and `Custom Callbacks` explaining how to add a custom callback (i.e. subclassing), not sure if showing an example is redundant with the other examples where we define a `PrintingCallback`.

Reply:
Oh yes that’s right. Maybe just link to that then?

Reply:
Actually nvm it’s the section right above anyways